src/InputGroup/index.js

```diff
@@ -156,126 +164,150 @@ StyledInputGroup.defaultProps = {
+  dataTest,
+  spaceAfter,
+  onFocus,
+  onBlur,
+  onChange,
+}: Props) => {
+  const [active, setActive] = useState(false);
+  const [filled, setFilled] = useState(false);
+  const [tooltipShown, setTooltipShown] = useState(false);
+  const [tooltipShownHover, setTooltipShownHover] = useState(false);
+  const labelRef = useRef(null);
+  const iconRef = useRef(null);
+  const inputID: string = useMemo(() => randomID("inputFieldID"), []);
+  const foundErrors = useMemo(() => findPropInChild("error", children), [children]);
+  const foundHelp = useMemo(() => findPropInChild("help", children), [children]);
+
+  const error = foundErrors.length > 0 && foundErrors[0];
+  const help = foundHelp.length > 0 && foundHelp[0];
+
+  const isFilled = useCallback(
+    () =>
+      setFilled(
+        !React.Children.map(
+          children,
+          child => child.props.value !== undefined && child.props.value !== "",
+        ).includes(false),
+      ),
+    [children],
+  );
+
+  useEffect(() => {
+    isFilled();
+  }, [isFilled, children]);
+
+  const handleFocus = useCallback(
+    (ev: SyntheticInputEvent<HTMLInputElement>) => {
+      setActive(true);
+      setTooltipShown(true);
+      if (onFocus) onFocus(ev);
+    },
+    [onFocus],
+  );
+
+  const handleBlur = useCallback(
+    (ev: SyntheticInputEvent<HTMLInputElement>) => {
+      isFilled();
+      setActive(false);
+      setTooltipShown(false);
+      if (onBlur) onBlur(ev);
+    },
+    [onBlur, isFilled],
+  );
+
+  const handleChange = useCallback(
+    (ev: SyntheticInputEvent<HTMLInputElement>) => {
+      isFilled();
+      if (onChange) {
+        onChange(ev);
+      }
+    },
+    [onChange, isFilled],
+  );
+
+  return (
+    <StyledInputGroup
+      label={label}
+      error={error}
+      active={active}
+      size={size}
+      dataTest={dataTest}
+      spaceAfter={spaceAfter}
+      role="group"
+      ariaLabelledby={label && inputID}
+      labelRef={label ? null : labelRef}
+    >
+      {label && (
+        <FormLabel
+          filled={filled}
+          id={inputID}
+          error={!!error}
+          help={!!help}
+          labelRef={labelRef}
+          iconRef={iconRef}
+          onMouseEnter={() => setTooltipShownHover(true)}
+          onMouseLeave={() => setTooltipShownHover(false)}
+        >
+          {label}
+        </FormLabel>
+      )}
+
+      <StyledChildren>
+        {React.Children.map(children, (item, key) => {
+          // either array, array with one length or string
+          // if it's not defined, use the first or string
+          const childFlex = Array.isArray(flex) && flex.length !== 1 ? flex[key] || flex[0] : flex;
+          return (
+            /* Until flow-bin@0.104.0 it's impossible to assign default values to
+            union types therefore, it's bypassed via declaring it here */
+            <StyledChild flex={childFlex || "0 1 auto"}>
+              {React.cloneElement(item, {
+                ref: node => {
```

Comment:
Please evaluate if the `item.ref` is not null, otherwise React can give us errors when the cloned element doesn't have `ref` 🙏